src/Kestrel.Core/KestrelServer.cs

```diff
@@ -135,7 +135,16 @@ private static ServiceContext CreateServiceContext(IOptions<KestrelServerOptions
 
                 async Task OnBind(ListenOptions endpoint)
                 {
-                    var connectionHandler = new ConnectionHandler<TContext>(endpoint, ServiceContext, application);
+                    // Add the connection limit middleware
+                    endpoint.UseConnectionLimit(ServiceContext);
+
+                    // Configure the user delegate
+                    endpoint?.Configure(endpoint);
```

Comment:
Well unit tests manually new up these things. I'll set the delegate in the ctor.